index.bs

```diff
@@ -790,7 +791,12 @@ To <dfn>initialize WebTransport over HTTP</dfn>, given a {{WebTransport}} object
   1. Let |connection| be the result of [=obtain a connection|obtaining a connection=] with
      |networkPartitionKey|, |url|, false, |newConnection|, and |http3Only|. If the user agent
      supports more than one congestion control algorithm, choose one appropriate for
-     |congestionControl| for sending of data on this |connection|.
+     |congestionControl| for sending of data on this |connection|. When obtaining a connection, if
+     |serverCertificateHashes| is specified, instead of using the default certificate verification
+     algorithm, consider the certificate valid if it meets the [=custom certificate
+     requirements=] and if [=verify a certificate hash|verifying the certificate hash=] against
+     |serverCertificateHashes| returns true. If either condition is not met, let |connection| be
+     failure.
```

Comment:
This bit about `serverCertificateHashes` might be better as a substep:

```
1. If |serverCertificateHashes| is specified, run the following steps:
    1.  If the certificate does not meet the [=custom certificate requirements=], mark the |connection| as failed.
    1. If the [=verify a certificate hash|verifying the certificate hash=] against |serverCertificateHashes| returns false, mark the |connection| as failed.
```

Reply:
I actually initially had this as substeps in earlier commits, but based on https://github.com/w3c/webtransport/pull/489#discussion_r1163085904, I changed it to be part of the same step. Since these steps are a modification of the "[obtain a connection](https://fetch.spec.whatwg.org/#concept-connection-obtain)" step in Fetch, I think we should keep this part of the same step so as to make it clear that these checks happen _during_ the "obtain a connection" step, as opposed to after.

Reply:
I don't think that conceptualizing these as part of "obtain a connection" helps.  It is cleaner if you regard these as checks that occur after a connection was successfully obtained.

Reply:
I slightly prefer keeping it part of the same step, but either seems workable, so I'll update it to whatever seems best to you and @vasilvv.

Reply:
> I don't think that conceptualizing these as part of "obtain a connection" helps. It is cleaner if you regard these as checks that occur after a connection was successfully obtained.

I'm not sure I understand why this would be a separate check, since presumably you would want to perform it at the place you would normally perform certificate validation (as soon as you receive a certificate), which is in the middle of obtaining a connection (e.g. there might be steps after, like sending H/3 settings; also from error handling perspective, that means that a failure against certificate hash list is equivalent to failure against Web PKI in terms of how it's reported to the application).

Reply:
It does seem like it ends up in a different subsystem and is something that is done during connection configuration at setup. 

In other words, in code, this isn't something that we check afterwards, but rather something that we configure as part of setting up a connection object and starting it.